pkg/types/validation/installconfig.go

```diff
@@ -99,6 +100,99 @@ func ValidateInstallConfig(c *types.InstallConfig, openStackValidValuesFetcher o
 	if _, ok := validPublishingStrategies[c.Publish]; !ok {
 		allErrs = append(allErrs, field.NotSupported(field.NewPath("publish"), c.Publish, validPublishingStrategyValues))
 	}
+
+	return allErrs
+}
+
+// IPAddressTypeForSections is a map of field type to whether they request IPv4 or IPv6.
+type IPAddressTypeForSections map[string]struct{ IPv4, IPv6 bool }
+
+// InferIPVersionFromInstallConfig infers the user's desired ip version from the networking config.
+// Presence field names match the field path of the struct within the Networking type. This function
+// assumes a valid install config.
+func InferIPVersionFromInstallConfig(n *types.Networking) (hasIPv4, hasIPv6 bool, presence IPAddressTypeForSections) {
+	if n == nil {
+		return
+	}
+	addresses := make(map[string][]net.IP)
+	for _, network := range n.MachineNetwork {
+		addresses["machineNetwork"] = append(addresses["machineNetwork"], network.CIDR.IP)
+	}
+	for _, network := range n.ServiceNetwork {
+		addresses["serviceNetwork"] = append(addresses["serviceNetwork"], network.IP)
+	}
+	for _, network := range n.ClusterNetwork {
+		addresses["clusterNetwork"] = append(addresses["clusterNetwork"], network.CIDR.IP)
+	}
+	presence = make(map[string]struct{ IPv4, IPv6 bool })
+	for k, ips := range addresses {
+		for _, ip := range ips {
+			has := presence[k]
+			if ip.To4() != nil {
+				has.IPv4 = true
+				hasIPv4 = true
+			} else {
+				has.IPv6 = true
+				hasIPv6 = true
+			}
+			presence[k] = has
+		}
+	}
+	return
+}
+
+// validateNetworkingIPVersion checks parameters for consistency when the user
+// requests single-stack IPv6 or dual-stack modes.
+func validateNetworkingIPVersion(c *types.InstallConfig) field.ErrorList {
+	var allErrs field.ErrorList
+
+	hasIPv4, hasIPv6, presence := InferIPVersionFromInstallConfig(c.Networking)
+
+	switch {
+	case hasIPv4 && hasIPv6:
+		switch {
+		case c.Platform.AWS != nil:
+		case c.Platform.Azure != nil:
+		case c.Platform.BareMetal != nil:
+		case c.Platform.None != nil:
```

Comment:
you can use https://github.com/openshift/installer/blob/master/pkg/types/installconfig.go#L157

Reply:
Is that easier to switch on?  Then I have two constants?